nujo/autodiff/tensor.py

```diff
@@ -191,7 +218,9 @@ def backward(self, _debug=False) -> None:
 
             if node.creator:
                 for child in node.creator.children:
-                    nodes_to_visit.insert(0, child)
+                    # Avoid visiting the same node twice
+                    if all(child is not node for node in nodes_to_visit):
```

Comment:
Can't this be done with 'child not in nodes_to_visit'?

Reply:
No, since `is in` compares by `==` (value) and here a comparison by `is` (address) is needed.